### DISCUSSION TITLE
How to make link prediction between nodes on unseen data

### DISCUSSION THREAD

**Discussion by Shee-Ra** [stellargraph/stellargraph#1838] [Q&A]
I'm new to stellar graph and am following the collab notebook _Link prediction with GraphSAGE_ [https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/graphsage-link-prediction.html](https://stellargraph.readthedocs.io/en/stable/demos/link-prediction/graphsage-link-prediction.html)

I have worked my way through the notebook, now let's suppose I want to put the model to work and make predictions where the outcome is unknown. How do I do that?

So if I have a new paper, how do I predict the citation links between this new paper node and the "citation graph".

Thanks.

**Comment by adocherty** [comment]
Hi, thanks for your question.

You can make predictions on nodes (labelled and unlabelled) using the `model.predict` method.

For example, in the "Link prediction with GraphSAGE" notebook, you can add the following line to make predictions on the test links, note that while the test links do have labels, the labels do not need to be supplied to the `flow` method, for example:

```
predict_flow = test_gen.flow(edge_ids_test)
test_predictions = model.predict(predict_flow)
```

I hope this helps, please let us know if you have further questions.
